<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SconeStudioClass</class>
 <widget class="QMainWindow" name="SconeStudioClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1077</width>
    <height>802</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SCONE</string>
  </property>
  <property name="toolButtonStyle">
   <enum>Qt::ToolButtonIconOnly</enum>
  </property>
  <property name="dockOptions">
   <set>QMainWindow::AllowTabbedDocks|QMainWindow::AnimatedDocks</set>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="spacing">
     <number>4</number>
    </property>
    <property name="leftMargin">
     <number>4</number>
    </property>
    <property name="topMargin">
     <number>4</number>
    </property>
    <property name="rightMargin">
     <number>4</number>
    </property>
    <property name="bottomMargin">
     <number>4</number>
    </property>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="viewerTab">
       <attribute name="title">
        <string>Viewer</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>4</number>
        </property>
        <property name="leftMargin">
         <number>4</number>
        </property>
        <property name="topMargin">
         <number>4</number>
        </property>
        <property name="rightMargin">
         <number>4</number>
        </property>
        <property name="bottomMargin">
         <number>4</number>
        </property>
        <item>
         <widget class="QOsgViewer" name="osgViewer" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>362</width>
            <height>361</height>
           </size>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QStackedWidget" name="stackedWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="currentIndex">
           <number>1</number>
          </property>
          <widget class="QWidget" name="playPage">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <property name="spacing">
             <number>4</number>
            </property>
            <property name="leftMargin">
             <number>4</number>
            </property>
            <property name="topMargin">
             <number>4</number>
            </property>
            <property name="rightMargin">
             <number>4</number>
            </property>
            <property name="bottomMargin">
             <number>4</number>
            </property>
            <item>
             <widget class="QPlayControl" name="playControl" native="true">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>24</height>
               </size>
              </property>
              <property name="autoFillBackground">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="progressPage">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <property name="spacing">
             <number>4</number>
            </property>
            <property name="leftMargin">
             <number>4</number>
            </property>
            <property name="topMargin">
             <number>4</number>
            </property>
            <property name="rightMargin">
             <number>4</number>
            </property>
            <property name="bottomMargin">
             <number>4</number>
            </property>
            <item>
             <widget class="QProgressBar" name="progressBar">
              <property name="value">
               <number>24</number>
              </property>
              <property name="textVisible">
               <bool>true</bool>
              </property>
              <property name="invertedAppearance">
               <bool>false</bool>
              </property>
              <property name="textDirection">
               <enum>QProgressBar::TopToBottom</enum>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="abortButton">
              <property name="text">
               <string>Abort</string>
              </property>
              <property name="checkable">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1077</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="action_Open_Scenario"/>
    <addaction name="action_Recent"/>
    <addaction name="separator"/>
    <addaction name="action_Close"/>
    <addaction name="separator"/>
    <addaction name="action_Save"/>
    <addaction name="actionSave_As"/>
    <addaction name="separator"/>
    <addaction name="actionE_xit"/>
   </widget>
   <widget class="QMenu" name="menu_View">
    <property name="title">
     <string>&amp;View</string>
    </property>
    <addaction name="actionShow_External_Forces"/>
    <addaction name="actionShow_Muscles"/>
    <addaction name="actionShow_Bone_Geometry"/>
    <addaction name="actionShow_Body_Axes"/>
    <addaction name="separator"/>
    <addaction name="actionUse_Log_Scale"/>
   </widget>
   <widget class="QMenu" name="menu_Optimize">
    <property name="title">
     <string>&amp;Optimize</string>
    </property>
    <addaction name="action_Optimize_Scenario"/>
    <addaction name="actionStart_Multiple_Optimizations"/>
    <addaction name="separator"/>
    <addaction name="actionTerminate_Running_Optimizations"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_View"/>
   <addaction name="menu_Optimize"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="resultsDock">
   <property name="minimumSize">
    <size>
     <width>160</width>
     <height>101</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>640</width>
     <height>524287</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Optimization Results</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="resultsDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <property name="spacing">
      <number>4</number>
     </property>
     <property name="leftMargin">
      <number>4</number>
     </property>
     <property name="topMargin">
      <number>4</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>4</number>
     </property>
     <item>
      <widget class="QFileBrowser" name="resultsBrowser">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="maximumSize">
        <size>
         <width>16777215</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Plain</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="messagesDock">
   <property name="minimumSize">
    <size>
     <width>80</width>
     <height>101</height>
    </size>
   </property>
   <property name="windowTitle">
    <string>Messages</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="messagesDockContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <property name="spacing">
      <number>4</number>
     </property>
     <property name="leftMargin">
      <number>4</number>
     </property>
     <property name="topMargin">
      <number>4</number>
     </property>
     <property name="rightMargin">
      <number>4</number>
     </property>
     <property name="bottomMargin">
      <number>4</number>
     </property>
     <item>
      <widget class="QLogSink" name="outputText">
       <property name="font">
        <font>
         <family>Consolas</family>
         <pointsize>9</pointsize>
        </font>
       </property>
       <property name="lineWrapMode">
        <enum>QPlainTextEdit::NoWrap</enum>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionE_xit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Alt+X</string>
   </property>
  </action>
  <action name="action_Open_Scenario">
   <property name="text">
    <string>&amp;Open Scenario...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_Recent">
   <property name="text">
    <string>&amp;Recent</string>
   </property>
  </action>
  <action name="action_Save">
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave_As">
   <property name="text">
    <string>Save &amp;As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="action_Close">
   <property name="text">
    <string>&amp;Close</string>
   </property>
  </action>
  <action name="action_Preferences">
   <property name="text">
    <string>&amp;Preferences...</string>
   </property>
  </action>
  <action name="action_About">
   <property name="text">
    <string>&amp;About...</string>
   </property>
  </action>
  <action name="action3D_Viewer">
   <property name="text">
    <string>3D &amp;Viewer</string>
   </property>
  </action>
  <action name="action_Run_Optimization">
   <property name="text">
    <string>&amp;Run Optimization</string>
   </property>
  </action>
  <action name="action_Optimize_Scenario">
   <property name="text">
    <string>Run &amp;Optimization</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F5</string>
   </property>
  </action>
  <action name="actionTerminate_Running_Optimizations">
   <property name="text">
    <string>&amp;Abort Optimizations...</string>
   </property>
  </action>
  <action name="action_Create_Video">
   <property name="text">
    <string>&amp;Create Video...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="actionStart_Multiple_Optimizations">
   <property name="text">
    <string>Run &amp;Multiple Optimizations...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+F5</string>
   </property>
  </action>
  <action name="actionShow_External_Forces">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show &amp;External Forces</string>
   </property>
  </action>
  <action name="actionShow_Muscles">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show &amp;Muscles</string>
   </property>
  </action>
  <action name="actionShow_Bone_Geometry">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show &amp;Bone Geometry</string>
   </property>
  </action>
  <action name="actionOptimization_Results">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Optimization &amp;Results</string>
   </property>
  </action>
  <action name="action_Messages">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Messages</string>
   </property>
  </action>
  <action name="actionShow_Body_Axes">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show Body &amp;Axes</string>
   </property>
  </action>
  <action name="actionUse_Log_Scale">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Use &amp;Log Scale with Optimization Progress</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="4" margin="4"/>
 <customwidgets>
  <customwidget>
   <class>QOsgViewer</class>
   <extends>QWidget</extends>
   <header>qosgviewer.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QPlayControl</class>
   <extends>QWidget</extends>
   <header>qplaycontrol.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>QFileBrowser</class>
   <extends>QTreeView</extends>
   <header>QFileBrowser.h</header>
  </customwidget>
  <customwidget>
   <class>QLogSink</class>
   <extends>QPlainTextEdit</extends>
   <header>qlogsink.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="SconeStudio.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>resultsBrowser</sender>
   <signal>activated(QModelIndex)</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>activateBrowserItem(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>183</x>
     <y>620</y>
    </hint>
    <hint type="destinationlabel">
     <x>183</x>
     <y>259</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Open_Scenario</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>fileOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Save</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>fileSave()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_As</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>fileSaveAs()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_About</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>helpAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>611</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Optimize_Scenario</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>optimizeScenario()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>611</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionTerminate_Running_Optimizations</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>abortOptimizations()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>611</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionE_xit</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>fileExit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>611</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Create_Video</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>createVideo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>611</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionStart_Multiple_Optimizations</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>optimizeScenarioMultiple()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>505</x>
     <y>357</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tabWidget</sender>
   <signal>tabCloseRequested(int)</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>tabCloseRequested(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>636</x>
     <y>298</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_External_Forces</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>updateViewSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_Muscles</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>updateViewSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_Bone_Geometry</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>updateViewSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>action_Preferences</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>showSettingsDialog()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>502</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>resultsDock</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>fixViewCheckboxes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>131</x>
     <y>291</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>messagesDock</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>fixViewCheckboxes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>538</x>
     <y>673</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_Body_Axes</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>updateViewSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionUse_Log_Scale</sender>
   <signal>triggered()</signal>
   <receiver>SconeStudioClass</receiver>
   <slot>updateViewSettings()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>538</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>activateBrowserItem(QModelIndex)</slot>
  <slot>updateScrollbar(int)</slot>
  <slot>start()</slot>
  <slot>stop()</slot>
  <slot>slomo(int)</slot>
  <slot>fileOpen()</slot>
  <slot>fileClose()</slot>
  <slot>fileSave()</slot>
  <slot>fileSaveAs()</slot>
  <slot>showViewer()</slot>
  <slot>viewAction()</slot>
  <slot>helpAbout()</slot>
  <slot>optimizeScenario()</slot>
  <slot>abortOptimizations()</slot>
  <slot>fileExit()</slot>
  <slot>selectBrowserItem(QModelIndex)</slot>
  <slot>createVideo()</slot>
  <slot>optimizeScenarioMultiple()</slot>
  <slot>tabCloseRequested(int)</slot>
  <slot>updateViewSettings()</slot>
  <slot>showSettingsDialog()</slot>
  <slot>viewResults(bool)</slot>
  <slot>viewMessages(bool)</slot>
  <slot>fixViewCheckboxes()</slot>
 </slots>
</ui>
